src/parquet/schema/printer.cc

```diff
@@ -93,16 +93,104 @@ static void PrintType(const PrimitiveNode* node, std::ostream& stream) {
   }
 }
 
+static void PrintLogicalType(LogicalType::type lt, std::ostream& stream) {
```

Comment:
The implementation is in 'src/parquet/types.cc'. Can you use that for your purpose ?